src/main/java/io/vertx/core/logging/Log4jLogDelegate.java

```diff
@@ -101,4 +150,8 @@ private void log(Level level, Object message) {
   private void log(Level level, Object message, Throwable t) {
     logger.log(FQCN, level, message, t);
   }
+
+  private void throwUnsupportedOperationException() {
```

Comment:
I'm not sure that this is acceptable, as it would avoid using the new methods for safety (As you don't know on which log system you are running). 

I would be in favor of computing the formatted message. This should check whether or not the log level is enabled before.